fastlane_bot/events/async_event_update_utils.py

```diff
@@ -100,20 +100,7 @@ def df(self) -> pd.DataFrame:
         """
         This property returns the response as a DataFrame.
         """
-        return pd.DataFrame(
-            [
-                {
-                    "exchange_name": self.exchange_name,
-                    "address": self.address,
-                    "tkn0_address": self.tkn0,
-                    "tkn1_address": self.tkn1,
-                    "fee": self.fee,
-                    "cid": self.cid,
-                    "strategy_id": self.strategy_id,
-                    "anchor": self.anchor,
-                }
-            ]
-        )
+        return pd.DataFrame([self.__dict__])
```

Comment:
wouldn't it be better to use `dataclass.asdict` here? that is imo the canonical way to converting a dataclass into a dict (importantly, it ignores non-dataclass properties should they ever be added)

Reply:
I had a similar thought about `__dict__` myself (more precisely, about the fact that it probably shouldn't be accessed directly). But I was unaware of the `asdict` option, which sounds like a right way to go about.

Reply:
I have changed to use asdict and also have added unit tests.